Although docs say CmdStan written w/o C++11, Makefiles require compiler w/ -std=c++1y flag

#### Summary:
The documentation appears to be out of date, and recommends a minimum compiler version that is too low to work with the current main Makefile.

#### Description:
The documentation for CmdStan 2.18 says, "CmdStan is written in portable C++ without C++11 features, as are the libraries on which it depends." It also says that on Linux, CmdStan can be compiled with a version of g++ that is "4.2.1 or later". However, the `CXXFLAGS` in the main Makefile include the option `-std=c++1y` by default, which not only implies that C++11 is needed, but also will not work with g++ 4.4.7, let alone the minimum version.

#### Reproducible Steps:

1. Have the default C++ compiler be g++ 4.4.7. This is already the case on Red Hat Enterprise Linux 6.10.

2. Untar cmdstan-2.18.0.tar.gz, and then `cd` into the resulting directory "cmdstan-2.18.0".

3. Type `make build` and press Enter.

#### Current Output:
```
g++ -Wall -I . -isystem stan/lib/stan_math/lib/eigen_3.3.3 -isystem stan/lib/stan_math/lib/boost_1.66.0 -isystem stan/lib/stan_math/lib/sundials_3.1.0/include -std=c++1y -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DBOOST_PHOENIX_NO_VARIADIC_EXPRESSION -Wno-unused-function -Wno-uninitialized -I src -isystem stan/src -isystem stan/lib/stan_math/ -DFUSION_MAX_VECTOR_SIZE=12 -Wno-unused-local-typedefs -DEIGEN_NO_DEBUG -DNO_FPRINTF_OUTPUT -pipe  -c src/cmdstan/stanc.cpp -O3 -o bin/cmdstan/stanc.o
cc1plus: error: unrecognized command line option "-std=c++1y"
cc1plus: warning: unrecognized command line option "-Wno-unused-local-typedefs"
make: *** [bin/cmdstan/stanc.o] Error 1
```

#### Expected Output:
I expect CmdStan to compile properly.

#### Current Version:
v2.18.0

--- a/src/docs/cmdstan-guide/installation.tex
+++ b/src/docs/cmdstan-guide/installation.tex
@@ -10,18 +10,17 @@ processor cores.
 
 \section{Operating System}
 
-\CmdStan is written in portable \Cpp without {\Cpp}11 features, as are the
+\CmdStan is written in portable \Cpp with {\Cpp}11 and {\Cpp}14 features, as are the
 libraries on which it depends.  Therefore, \CmdStan should run on any machine
-for which a suitable \Cpp compiler is available.  In practice, \CmdStan,
-like the Boost and Eigen libraries on which it depends, is very hard
-on the compiler and linker.
+for which a suitable \Cpp compiler supporting C++1y or C++14 features is available. In practice, \CmdStan,
+like the Boost and Eigen libraries on which it depends, is very hard on the compiler and linker.
 
 \CmdStan has been tested on the following operating systems.
 %
 \begin{itemize}
-\item Linux (Debian, Ubuntu, Red Hat)
+\item Linux (Debian, Ubuntu)
 \item Mac OS X (from 10.6 ``Snow Leopard''  through 10.14 ``Mojave'')
-\item Windows (XP, 7, 8).
+\item Windows (7, 8, 10).
 \end{itemize}
 %
 \CmdStan should work on other versions of these operating systems if
@@ -333,7 +332,7 @@ To test, open the Terminal application and enter
 \end{quote}
 %
 Verify that \code{make} is at version 3.81 or later and \code{g++}
-is at 4.2.1 or later.
+is at 4.9.3 or later.
 
 
 \subsection{Download and Unpack \CmdStan Source}
@@ -379,18 +378,18 @@ are already installed, use the commands
 \end{Verbatim}
 \end{quote}
 %
-If these are at least at \code{g++} version 4.2.1 or later and
+If these are at least at \code{g++} version 4.9.3 or later and
 \code{make} version 3.81 or later, no additional installations are
 necessary.  It may still be desirable to update the \Cpp compiler
 \code{g++}, because later versions are faster.
 
-To install the latest version of these
-tools (or upgrade an older version), use the commands
+To install the latest version of these 
+tools (or upgrade an older version), use the following commands or their equivalent for your distribution: 
 %
 \begin{quote}
 \begin{Verbatim}[fontshape=sl,fontsize=\small]
-> sudo apt-get install g++
-> sudo apt-get install make
+> sudo apt install g++
+> sudo apt install make
 \end{Verbatim}
 \end{quote}
 %

